haas/client.py

```diff
@@ -0,0 +1,801 @@
+# Copyright 2013-2015 Massachusetts Open Cloud Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the
+# License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS
+# IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied.  See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""This module implements the HaaS client library."""
+from haas import config
+from haas.config import cfg
+from error_clientlib import *
+
+from haas.rest import APIError, ServerError
+    # Do we need ServerError ? It is not supposed to know 
+    # what happens on the server side
+import requests
+import os
+import urllib
+import json
+
+
+#Return Codes:
+""" Following is how return codes should be interpreted:
+    -1  :       Operation Failed
+    1   :       Operation was successful
+    2   :       Duplication error. Entity already exists
+    3   :
+
+"""
+
+
+
+class Haas:
+    """Client class for making Haas API calls.
+    Note that this client library is not yet complete and will be added to as
+    needed.
+
+    Example:
+        h = Haas(endpoint="http://127.0.0.1:5000")
+        h.node_connect_network("node-2000", "eth1", "production-network",
+        "vlan/native")
+
+    Errors are thrown when receiving HTTP status_codes from the HaaS server
+    that are not in the [200,300) range
+    """
+
+    def __init__(self, endpoint=None):
+        """Initiatlize an instance.
+
+        If endpoint is None, use the endpoint specification from:
+            1) The HAAS_ENDPOINT env variable or
+            2) Take the endpoint from [client] endpoint
+
+        Exceptions:
+            LookupError - no endpoint could be found
+        """
+        config.load(filename='haas.cfg')
+        if endpoint != None:
```

Comment:
We're not preferring HAAS_ENDPOINT over the config file like the comment above specifies.